pkg/splunk/client/awss3client.go

```diff
@@ -52,23 +51,10 @@ type AWSS3Client struct {
 	AWSSecretAccessKey string
 	Prefix             string
 	StartAfter         string
-	Endpoint           string
 	Client             SplunkAWSS3Client
 	Downloader         SplunkAWSDownloadClient
 }
 
-// regex to extract the region from the s3 endpoint
```

Comment:
Rather than completely removing the extraction, should we still maintain backward compatibility by attempting extraction if region in the CR spec hasn't been specified?

Otherwise, customers upgrading to this version without region explicitly set in their CRs would start seeing failures?